### PR TITLE
StratifiedSample software eng edits, reorder, testing

### PR DESCRIPTION
**Pull request by smishr**
Similar to the other PR for `SimpleRandomSample`, incorporates type checking and better if-else ladder for `StratifiedSample`

### CODE REVIEW

**Comment by smishr** [comment]
Have overhauled the src of StratifiedSample, still have to write correct testing suite

File: src/SurveyDesign.jl [hunk 1]
```diff
@@ -178,75 +179,158 @@ end
     StratifiedSample <: AbstractSurveyDesign
 
     Survey design sampled by stratification.
+
```
Comment (ayushpatnaikgit) [inline src/SurveyDesign.jl:182]: I think the doc should have an example.

Comment (smishr) [reply src/SurveyDesign.jl:182]: okay. ill add

Comment (smishr) [reply src/SurveyDesign.jl:182]: how many lines of output must be copied into doctring?

Comment (smishr) [reply src/SurveyDesign.jl:204]: I have added and fixed the jldoctests. They have to be spaced right to margin, no tabs before to be displayed like a code block

**Pull request merged**